ratpack-core/src/main/java/ratpack/http/client/internal/DefaultHttpClient.java

```diff
@@ -279,27 +286,33 @@ public HttpClientSpec enableMetricsCollection(boolean enableMetricsCollection) {
   public Promise<ReceivedResponse> request(URI uri, final Action<? super RequestSpec> requestConfigurer) {
     return intercept(
       Promise.async(downstream -> new ContentAggregatingRequestAction(uri, this, 0, Execution.current(), requestConfigurer.append(spec.requestInterceptor)).connect(downstream)),
-      spec.responseInterceptor
+      spec.responseInterceptor,
+      spec.errorInterceptor
     );
   }
 
   @Override
   public Promise<StreamedResponse> requestStream(URI uri, Action<? super RequestSpec> requestConfigurer) {
     return intercept(
       Promise.async(downstream -> new ContentStreamingRequestAction(uri, this, 0, Execution.current(), requestConfigurer.append(spec.requestInterceptor)).connect(downstream)),
-      spec.responseInterceptor
+      spec.responseInterceptor,
+      spec.errorInterceptor
     );
   }
 
-  private <T extends HttpResponse> Promise<T> intercept(Promise<T> promise, Action<? super HttpResponse> action) {
-    return promise.next(r ->
-      ExecController.require()
-        .fork()
-        .eventLoop(Execution.current().getEventLoop())
-        .start(e ->
-          action.execute(r)
-        )
-    );
+  private <T extends HttpResponse> Promise<T> intercept(Promise<T> promise, Action<? super HttpResponse> action, Action<? super Throwable> errorAction) {
+    return promise.onError(Exception.class, t -> {
```

Comment:
I don't like the use of `Exception` here, but `Action` only allows `Exception` rather than `Throwable` to be thrown, even though `#onError` catches `Throwable`.

Reply:
You can use `Promise.wireTap(Action<Result>)` in order to register an error action that is not inline with the promise chain execution. This will allow you to not have to handle and re-throw the Throwable.

We probably should also execute the `errorAction` in a forked execution just like with do with the response interceptor. That way you can add blocking code without impacting the main program flow.